qt_gui/CMakeLists.txt

```diff
@@ -11,13 +11,12 @@ install(DIRECTORY resource
   DESTINATION share/${PROJECT_NAME})
 
 if(BUILD_TESTING)
-  find_package(ament_cmake_pytest REQUIRED)
   find_package(ament_lint_auto REQUIRED)
-  ament_lint_auto_find_test_dependencies()
 
-  ament_add_pytest_test(${PROJECT_NAME} test
-    APPEND_ENV PYTHONPATH=${CMAKE_CURRENT_BINARY_DIR}
-    TIMEOUT 90)
+  # Disabling copyright test. The copyright used in this package does not conform to
+  # ament's copyright tests
+  set(ament_cmake_copyright_FOUND TRUE)
+  ament_lint_auto_find_test_dependencies()
```

Comment:
This test seems really really picky. @dirk-thomas is it possible to pass in flags for flake8 rules to ignore such as Q000?

Reply:
No, the linters don't have many arguments. The idea is that they define the code style of ROS 2. If every package would customize it the result wouldn't be consistent. So each package should decide if it wants to follow the ROS 2 code style (and be able to use its linters) or not (and optionally do any custom linting on its own).

I think the only option changable is extending the max line length since for generated code we simply can't guarantee an upper bound like 80 / 100 / 120.

The Q000 warning of `flake8-quotes` is explicitly there: see https://index.ros.org/doc/ros2/Developer-Guide/#python We think having a deterministic way to write a string literal is beneficial - not only for readability / consistency but also for searchability.

Reply:
As an interesting side note: this is actually how Python quotes string. From the `PyString_Repr` function:

```
/* figure out which quote to use; single is preferred */
quote = '\'';
if (smartquotes &&
    memchr(op->ob_sval, '\'', Py_SIZE(op)) &&
    !memchr(op->ob_sval, '"', Py_SIZE(op)))
    quote = '"';
```

Reply:
Search-ability is a good argument. I'm all for standardizing ros2 style for new packages but when porting from ROS1, forcing all of flake8 borders on being so strict that users may decide to not to include this test. For example, some of the rules about docstrings are really really overkill and abiding by them is a lot of overhead. For instance, ros2cli must not be running this test as it violates a lot of the flake8 rules